crates/iceberg/src/expr/visitors/manifest_evaluator.rs

```diff
@@ -221,67 +394,213 @@ impl ManifestFilterVisitor<'_> {
         let pos = reference.accessor().position();
         &self.partitions[pos]
     }
+
+    #[allow(clippy::bool_comparison)]
```

Comment:
Similar to prev comment, please refactor to avoid needing to disable this clippy lint.